xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -170,48 +181,137 @@ public void handleNameResolutionError(Status error) {}
 
             @Override
             public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
-              if (xdsClient == null) {
-                ManagedChannel channel = initLbChannel(helper, newBalancerName, channelCredsList);
-                xdsClient  = new XdsComms2(
-                    channel, helper, new ExponentialBackoffPolicy.Provider(),
-                    GrpcUtil.STOPWATCH_SUPPLIER, node);
-                LoadStatsStore loadStatsStore = new LoadStatsStoreImpl();
-                localityStore = localityStoreFactory.newLocalityStore(
-                    helper, lbRegistry, loadStatsStore);
+              if (xdsClientRef != null) {
+                return;
+              }
+
+              LoadStatsStore loadStatsStore = new LoadStatsStoreImpl();
+              localityStore = localityStoreFactory.newLocalityStore(
+                  helper, lbRegistry, loadStatsStore);
+              // TODO(zdapeng): Use XdsClient to do Lrs directly.
+              final LoadReportCallback lrsCallback =
+                  new LoadReportCallback() {
+                    @Override
+                    public void onReportResponse(long reportIntervalNano) {
+                      localityStore.updateOobMetricsReportInterval(reportIntervalNano);
+                    }
+                  };
+
+              if (newXdsConfig.balancerName != null) {
+                // The is to handle the legacy usecase that requires balancerName from xds config.
+
+                String oldBalancerName = oldXdsConfig == null ? null : oldXdsConfig.balancerName;
+                if (!newXdsConfig.balancerName.equals(oldBalancerName)) {
+                  final ManagedChannel channel = initLbChannel(
+                      helper, newXdsConfig.balancerName, Collections.<ChannelCreds>emptyList());
+                  final Node node = Node.newBuilder()
+                      .setMetadata(Struct.newBuilder()
+                          .putFields(
+                              "endpoints_required",
+                              Value.newBuilder().setBoolValue(true).build()))
+                      .build();
+                  // TODO(zdapeng): Use XdsClient to do Lrs directly.
+                  lrsClient = loadReportClientFactory.createLoadReportClient(
+                      channel, helper, new ExponentialBackoffPolicy.Provider(), loadStatsStore);
+                  xdsClientRef = new RefCountedXdsClientObjectPool(new XdsClientFactory() {
+                    @Override
+                    XdsClient createXdsClient() {
+                      return new XdsComms2(
```

Comment:
Why not use `XdsClientImpl` or a comment about why still `XdsComm2`?

Reply:
EDS part for `XdsClientImpl` is not merged yet. So it does not do EDS now. After that gets merged, we can get rid of `XdsComms2`.